### PR TITLE
TRUNK-2497: Store outgoing MessageService emails in a table

### PR DESCRIPTION
**Pull request by lehone-hp**
## Description of what I changed

* Create domain object SentMessage
* Added table sent_message with columns sender (a user_id), recipient (a user_id), message, date_sent, type_of_message
* Created  a new hibernate mapping hbm.xml file for this new object to be able to persist the messages in the archive.
* Add methods to the API/DAO for CRUD operations for sent messages
* Include unit tests

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-2497

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/resources/org/openmrs/notification/db/hibernate/MessageReceiver.hbm.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	<class name="org.openmrs.notification.MessageReceiver" table="message_receiver">
+		<id name="id" type="int" column="id" unsaved-value="0">
+			<generator class="native">
+				<param name="sequence">message_receivers_id_seq</param>
+			</generator>
+		</id>
+		<discriminator column="id" insert="false" />
+		<many-to-one name="sentMessage" column="message_id" class="org.openmrs.notification.SentMessage" not-null="true" />
+		<many-to-one name="receiver" column="receiver_id" class="org.openmrs.User" not-null="true" />
+		<property name="uuid" column="uuid" type="java.lang.String" unique="true" />
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/notification/db/hibernate/MessageReceiver.hbm.xml]: Would you like the uuid to be nullable? Does it take on any length?

File: api/src/main/resources/org/openmrs/notification/db/hibernate/SentMessage.hbm.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	<class name="org.openmrs.notification.SentMessage" table="sent_message">
+		<id name="messageId" type="int" column="message_id" unsaved-value="0">
+			<generator class="native">
+				<param name="sequence">sent_messages_message_id_seq</param>
+			</generator>
+		</id>
+		<discriminator column="message_id" insert="false" />
+		<many-to-one name="sender" column="sender_id" class="org.openmrs.User" not-null="true" />
+		<property name="uuid" column="uuid" type="java.lang.String" unique="true" />
```
Comment (dkayiwa) [inline api/src/main/resources/org/openmrs/notification/db/hibernate/SentMessage.hbm.xml]: Same as above

File: api/src/main/resources/org/openmrs/notification/db/hibernate/MessageReceiver.hbm.xml (hunk 1 above)
Comment (lehone-hp) [reply api/src/main/resources/org/openmrs/notification/db/hibernate/MessageReceiver.hbm.xml]: No, Let me fix that

File: api/src/main/resources/org/openmrs/notification/db/hibernate/SentMessage.hbm.xml (hunk 2 above)
Comment (lehone-hp) [reply api/src/main/resources/org/openmrs/notification/db/hibernate/SentMessage.hbm.xml]: I'll fix this as well

**Comment by dkayiwa** [comment]
Did you test these changes by running the web application to check if all is well?

**Comment by lehone-hp** [comment]
Yes I did and and is well

**Comment by dkayiwa** [comment]
@lehone-hp are you still interested in finishing this up?

**Pull request closed without merging**